### PR TITLE
Array - matrix diagonal

### PR DESCRIPTION
**Pull request by DevratAgarwal**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [x] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [ ] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Comment by sailesh-ramayanam** [comment]
@satya-9 
1. Variable names in the template code can be better
2. Strings are not converted to integers

File: solutions/arrays/matrix-diagonal.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+// name your function as change_diagonal and it should take list as input
+function change_diagonal(arr)
+{
+	for(let row=0;row<arr.length;row++)
+	{
+		for(let column=0;column<arr.length;column++)
```
Comment (sailesh-ramayanam) [inline solutions/arrays/matrix-diagonal.js:18]: @DevratAgarwal This is O(n^2) time. We don't need a nested loop. We can do in O(n) time.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

**Comment by DevratAgarwal** [comment]
@sailesh-ramayanam @satya-9 changes are addressed

**Comment by LVSsandeepVarma** [comment]
> 

@sailesh-ramayanam  , @DevratAgarwal  updated the template code

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**